data/trie/sync.go

```diff
@@ -244,7 +273,16 @@ func (ts *trieSyncer) getNode(hash []byte) (node, error) {
 		return trieNode(n)
 	}
 
-	return nil, ErrNodeNotFound
+	existingNode, err := getNodeFromDBAndDecode(hash, ts.trie.Database(), ts.trie.marshalizer, ts.trie.hasher)
+	if err != nil {
+		return nil, ErrNodeNotFound
+	}
+	err = existingNode.setHash()
```

Comment:
why is setHash needed ?

Reply:
when loading a new node from storage the hash field is not set. This was observed during testing.